tests/typechecking/decorators/test_superfunction_typing.py

```diff
@@ -0,0 +1,97 @@
+
+import pytest
+import asyncio
+import sys
+from contextlib import suppress
+
+if sys.version_info <= (3, 11):
+    from typing_extensions import reveal_type
+else:
+    from typing import reveal_type
+
+from transfunctions import superfunction, sync_context, async_context
+
+
+@superfunction
+def typed_superfunction(arg: float, *, kwarg: int = 0) -> int:
+    with sync_context:
+        return 1
+    with async_context:
+        return 2
+    # TODO: add test case for generator_context once there is a typing solution
```

Comment:
If it doesn't work, it's better to make a test function with this and temporary skip it. This is the fastest way to implement it in the future.